src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -603,7 +590,14 @@ public CompilationInfo(Compilation compilation, bool hasCompleteReferences)
                 try
                 {
                     bool hasCompleteReferences = true;
+
                     var newReferences = new List<MetadataReference>();
+
+                    // we don't consider metadata reference being actually exist on disk as a part of complete reference
+                    // since we know some projects do have those but many just work fine since those dlls are not actually used.
```

Comment:
The compiler already has to do that check once it's actually fetching metadata. Can we take advantage of that check?